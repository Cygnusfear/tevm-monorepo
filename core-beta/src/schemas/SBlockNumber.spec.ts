import { type BlockNumber, parseBlockNumber } from './SBlockNumber'
import { assertType, describe, expect, it } from 'vitest'

describe(parseBlockNumber.name, () => {
	it('should return the validated block number', () => {
		const expected = 5 satisfies BlockNumber
		const bn = parseBlockNumber(expected) satisfies BlockNumber
		assertType<typeof expected>(bn)
		expect(parseBlockNumber(0)).toBe(0)
		expect(parseBlockNumber(5)).toBe(5)
		expect(parseBlockNumber(90000000000)).toBe(90000000000)
	})

	it('should throw if invalid address', () => {
		expect(() => parseBlockNumber(52n as any)).toThrowErrorMatchingSnapshot()
		expect(() => parseBlockNumber('0x' as any)).toThrowErrorMatchingSnapshot()
		expect(() => parseBlockNumber('0' as any)).toThrowErrorMatchingSnapshot()
		expect(() => parseBlockNumber('' as any)).toThrowErrorMatchingSnapshot()
		expect(() => parseBlockNumber(true as any)).toThrowErrorMatchingSnapshot()
		expect(() => parseBlockNumber({} as any)).toThrowErrorMatchingSnapshot()
		expect(() =>
			parseBlockNumber('not an address' as any),
		).toThrowErrorMatchingSnapshot()
	})
})
