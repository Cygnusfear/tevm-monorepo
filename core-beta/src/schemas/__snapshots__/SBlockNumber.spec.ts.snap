// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseBlockNumber > should throw if invalid address 1`] = `
"InvalidBlockNumberError: Value 52 is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;

exports[`parseBlockNumber > should throw if invalid address 2`] = `
"InvalidBlockNumberError: Value 0x is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;

exports[`parseBlockNumber > should throw if invalid address 3`] = `
"InvalidBlockNumberError: Value 0 is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;

exports[`parseBlockNumber > should throw if invalid address 4`] = `
"InvalidBlockNumberError: Value  is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;

exports[`parseBlockNumber > should throw if invalid address 5`] = `
"InvalidBlockNumberError: Value true is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;

exports[`parseBlockNumber > should throw if invalid address 6`] = `
"InvalidBlockNumberError: Value [object Object] is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;

exports[`parseBlockNumber > should throw if invalid address 7`] = `
"InvalidBlockNumberError: Value not an address is not a valid BlockNumber. BlockNumbers must be an integer >= 0
https://evmts.dev/reference/errors"
`;
